### PR TITLE
JSON support, first cut

### PR DESCRIPTION
**Pull request by hemanth**
Potential fix for #10 

It would be better to mandate default, so that ext might be optional?

### CODE REVIEW

File: configstore.js [hunk 1]
```diff
@@ -23,24 +24,11 @@ Configstore.prototype = Object.create(Object.prototype, {
...
-				}
-
-				// improve the message of permission errors
-				if (err.code === 'EACCES') {
-					err.message = err.message + '\n' + permissionError + '\n';
-				}
-
-				// empty the file if it encounters invalid YAML
-				if (err.name === 'YAMLException') {
+				return require(this.path);;
```
Comment (arthurvr) [inline configstore.js]: two semicolons lol

File: configstore.js [hunk 2]
```diff
@@ -53,10 +41,7 @@ Configstore.prototype = Object.create(Object.prototype, {
 				// make sure the folder exists, it could have been
 				// deleted meanwhile
 				mkdirp.sync(path.dirname(this.path));
-				fs.writeFileSync(this.path, yaml.safeDump(val, {
-					skipInvalid: true,
-					schema: yaml.JSON_SCHEMA
-				}));
+				fs.writeFileSync(this.path, JSON.stringify(val, null, 4))
```
Comment (arthurvr) [inline configstore.js]: and this misses one :)

File: configstore.js [hunk 3]
```diff
@@ -23,24 +24,11 @@ Configstore.prototype = Object.create(Object.prototype, {
...
-				// improve the message of permission errors
-				if (err.code === 'EACCES') {
-					err.message = err.message + '\n' + permissionError + '\n';
-				}
-
-				// empty the file if it encounters invalid YAML
-				if (err.name === 'YAMLException') {
+				return require(this.path);;
+			}
+		    catch (err) {
```
Comment (arthurvr) [inline configstore.js]: this indentation is screwed up

**Comment by arthurvr** [comment]
@sindresorhus @hemanth Where did we end up with this? It needs a rebase and tests seem to fail.

> hence the 2.0. We should add a warning that you shouldn't upgrade to 2.0, but rather only use it if you use it from scratch.

:+1: That message will need to be added as well.

**Comment by sindresorhus** [comment]
And:

> As commented in hangouts, I don't want more options.

File: configstore.js [hunk 4]
```diff
@@ -23,24 +24,11 @@ Configstore.prototype = Object.create(Object.prototype, {
...
-				// create dir if it doesn't exist
-				if (err.code === 'ENOENT') {
-					mkdirp.sync(path.dirname(this.path));
-					return {};
-				}
-
-				// improve the message of permission errors
-				if (err.code === 'EACCES') {
-					err.message = err.message + '\n' + permissionError + '\n';
-				}
```
Comment (sindresorhus) [inline configstore.js]: ??

File: configstore.js [hunk 5]
```diff
@@ -23,24 +24,11 @@ Configstore.prototype = Object.create(Object.prototype, {
...
-				return yaml.safeLoad(fs.readFileSync(this.path, 'utf8'), {
-					filename: this.path,
-					schema: yaml.JSON_SCHEMA
-				});
-			} catch (err) {
-				// create dir if it doesn't exist
-				if (err.code === 'ENOENT') {
-					mkdirp.sync(path.dirname(this.path));
-					return {};
-				}
```
Comment (sindresorhus) [inline configstore.js]: ??

File: configstore.js (hunk 1 above)
Comment (sindresorhus) [inline configstore.js]: use normal `fs.readFileSync` and `JSON.parse`

**Comment by hemanth** [comment]
Thanks @sindresorhus :)

**Pull request closed without merging**